components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/util/TemplateManagerHelper.java

```diff
@@ -454,7 +514,7 @@ public static String replaceRegex(String stringWithRegex, String regexPatternStr
             }
             return variableValues;
         } catch (ScriptException e) {
-            throw new RuleTemplateScriptException("Error occurred while running the script", e);
+            throw new RuleTemplateScriptException(e.getCause().getMessage(), e);
```

Comment:
Let's provide a more descriptive error message. Since stactTrace is provided, no need to provide e.getCause().getMessage() for the custom exception.

Reply:
A RuleTemplateScriptException occurs when executing the JavaScript fails (either a validation for a user-entered value fails, or the script has errors). e.getCause().getMessage() gives that particular error. Stacktrace identifies the complete error including Nashorn engine's ScriptException

Reply:
+1 for informing the exact cause for the failure.
But If you are going with this approach, at the top level you should provide related details to the failure such as respective RuleTemplateID etc.. in logs.

Reply:
Yes, anyhow the script executes within a rule template. A script failure means that 
- A script in a rule template is invalid - It falls under the log of rule template validation method, and then to the respective template group validation method.
or
- Value entered by the user in the front end has thrown a validation error from the script, when trying to process it - Error from the script is shown to the user as an error message, referring to the entered value (eg: "A Number is required instead of the entered value - 'foo' "). So, the user can change it